similar/main/state.cpp

```diff
@@ -1536,6 +1536,12 @@ int state_restore_all_sub(const char *filename, const secret_restore secret)
 				do_cloak_invul_secret_stuff(old_gametime, ret_pl_info);
 			} else {
 				get_local_player() = dummy_player;
+				// Keep keys even if they died on secret level (otherwise game becomes impossible)
+				// Example: Cameron 'Stryker' Fultz's Area 51
+				pl_info.powerup_flags |= (plrobj.ctype.player_info.powerup_flags &
```

Comment:
You're right, I had a copy of 0.57.3 on hand and it doesn't keep any keys I got on a secret level if I die on the secret level. However, it does keep the keys if I die on a normal level.

I would argue that this was overlooked by the original programmers simply because none of the secret levels in Descent II: Counterstrike have keys in them. It is possible to win Area 51, you just have to make sure you don't die between getting a key on the secret level and returning to the base level. However, if you do die in that situation, even if it's not Game Over, any keys you picked up in the secret level in that session of playing it are permanently lost, making the mission unwinnable. Quite frankly, I think this was overlooked by the mission creator. He made a note of getting all the keys before blowing up the secret level, but not of this possibility I described.